### PR TITLE
refactor: improved drag placement behavior

### PR DESCRIPTION
**Pull request by btn0s**
fixed acelleration on drag

increased drag detection threshold

### CODE REVIEW

File: components/Block.vue [hunk 1]
```diff
@@ -75,66 +75,56 @@ export default {
...
       }
 
-      this.dragStartCoords.x = clientX
-      this.dragStartCoords.y = clientY
       this.isDragging = true
     },
     handleDrag(e) {
-      e.preventDefault()
       const { clientX, clientY } = e
+      const boxNode = e.target.parentNode.parentNode.getBoundingClientRect()
```
Comment (btn0s) [inline components/Block.vue]: I guess here I could just collapse these two lines

File: components/Block.vue [hunk 2]
```diff
@@ -75,66 +75,56 @@ export default {
...
-        height: `${offsetHeight}px`,
-        width: `${offsetWidth}px`,
+        height: `${height}px`,
+        width: `${width}px`,
         top: `${clientY}px`,
         left: `${clientX}px`,
         transform: `translate(${-Math.abs(offsetX)}px,${-Math.abs(offsetY)}px)`
       }
 
-      this.dragStartCoords.x = clientX
```
Comment (btn0s) [inline components/Block.vue:88]: These are still in the local state, should probably be removed

File: components/Block.vue [hunk 3]
```diff
@@ -229,6 +219,7 @@ export default {
 <style>
 .Block {
   @apply relative;
+  user-select: none;
```
Comment (btn0s) [inline components/Block.vue:222]: this is unrelated but wanted to put it in to remove some funky selection behavior

File: components/Block.vue [hunk 4]
```diff
@@ -75,66 +75,56 @@ export default {
...
         if (this.blockData.y > 0) {
           this.blockData.y -= 1
         }
       }
     },
-    handleDragEnd(e) {
-      e.preventDefault()
-      this.dragStartCoords = { x: 0, y: 0 }
-      this.ghostDraggingStyle = {}
+    handleDragEnd() {
```
Comment (juanreyes1212) [inline components/Block.vue:125]: I'm thinking the conditionals here might be what's giving you the janky stuff on resizing. I want to play with this haha

**Review by juanreyes1212 — APPROVED** [review]

File: components/Block.vue (hunk 4 above)
Comment (juanreyes1212) [reply components/Block.vue:125]: oops meant the (x, y) subtractions/updates

Comment (btn0s) [reply components/Block.vue:125]: pull it down and play with it! i might actually end up using this for the admin side of another project

File: components/Block.vue [hunk 5]
```diff
@@ -75,66 +74,57 @@ export default {
     },
     handleDragStart(e) {
       const { clientX, clientY, offsetX, offsetY } = e
-      const { offsetHeight, offsetWidth } = e.target.offsetParent.parentElement
+      const {
+        height,
+        width
+      } = e.target.parentNode.parentNode.getBoundingClientRect()
```
Comment (seanmcoleman) [inline components/Block.vue:80]: I don't know what the DOM structure looks like but `parentNode.parentNode` seems brittle. Is there a more direct reference you can get?

File: components/Block.vue [hunk 6]
```diff
@@ -75,66 +74,57 @@ export default {
...
       }
 
-      this.dragStartCoords.x = clientX
-      this.dragStartCoords.y = clientY
       this.isDragging = true
     },
     handleDrag(e) {
-      e.preventDefault()
       const { clientX, clientY } = e
+      const { x, y } = e.target.parentNode.parentNode.getBoundingClientRect()
```
Comment (seanmcoleman) [inline components/Block.vue:94]: There's definitely duplication here that could probably be extracted to some computed property or something.

File: components/Block.vue [hunk 7]
```diff
@@ -75,66 +74,57 @@ export default {
...
 
-      const movementX = clientX - this.dragStartCoords.x
-      const movementY = clientY - this.dragStartCoords.y
-
-      if (movementX > 40) {
-        this.dragStartCoords.x = clientX
+      const movementX = clientX - x
+      const movementY = clientY - y
 
+      if (movementX > 70) {
```
Comment (seanmcoleman) [inline components/Block.vue:105]: I know these values were hardcoded before but could be useful as constants.

File: components/Block.vue (hunk 5 above)
Comment (btn0s) [reply components/Block.vue:80]: I don't think there's a better way to get the grandparent through the event, I could just throw an ID on the parent and grab it that way with `.getElementById(id)` I guess

**Pull request closed without merging**